kv/kvserver: TestClosedTimestampFrozenAfterSubsumption failed

kv/kvserver.TestClosedTimestampFrozenAfterSubsumption [failed](https://tanzanite.cluster.engflow.com/invocations/default/8a7c12c7-d20b-467a-aacf-c88976f63811?testReportRun=2&testReportShard=42&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.1 @ [38229635f670bbdd2b22640a55a076d3ae373ead](https://github.com/cockroachdb/cockroach/commits/38229635f670bbdd2b22640a55a076d3ae373ead):


```
=== RUN   TestClosedTimestampFrozenAfterSubsumption
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestClosedTimestampFrozenAfterSubsumption3373369037
    test_log_scope.go:76: use -show-logs to present logs inline
    closed_timestamp_test.go:849: -- test log scope end --
test logs left over in: outputs.zip/logTestClosedTimestampFrozenAfterSubsumption3373369037
--- FAIL: TestClosedTimestampFrozenAfterSubsumption (92.64s)
=== RUN   TestClosedTimestampFrozenAfterSubsumption/rhs_lease_transfer_while_subsumed
    closed_timestamp_test.go:835: 
        	Error Trace:	pkg/kv/kvserver/closed_timestamp_test.go:835
        	Error:      	Should be true
        	Test:       	TestClosedTimestampFrozenAfterSubsumption/rhs_lease_transfer_while_subsumed
    --- FAIL: TestClosedTimestampFrozenAfterSubsumption/rhs_lease_transfer_while_subsumed (87.66s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=42</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClosedTimestampFrozenAfterSubsumption.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47093

--- a/pkg/kv/kvserver/closed_timestamp_test.go
+++ b/pkg/kv/kvserver/closed_timestamp_test.go
@@ -625,6 +625,7 @@ func TestClosedTimestampFrozenAfterSubsumption(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t)
+	skip.UnderDeadlock(t)
 
 	for _, test := range []struct {
 		name string

